### ISSUE TITLE
'next' acts like 'continue' with mechanize

### ISSUE THREAD

**Issue by timscott** [deivid-rodriguez/byebug#103]
I've had this issue for a while, and I just live with it by making sure to stop before each mechanize call and put a breakpoint right after it. As you can imagine, it's quite painful.

This shows the problem.:

```ruby
require 'mechanize'
agent = Mechanize.new
agent.get 'http://www.google.com'
puts "When I tell byebug 'next' from the line above, I expect to stop before executing this line, but it does not."
puts "If fact it doesn't stop here either. It acts as though I asked to 'continue'. Whatddup?"
```

**Comment by deivid-rodriguez** [comment]
Hi @timscott, I've further isolated this to

    require 'domain_name'
    domain = DomainName.new('google.com')
puts "When I tell byebug 'next' from the line above, I expect to stop before executing this line, but it does not."

It seems like a bug in the TracePoint API, so I've reported to ruby-core: https://bugs.ruby-lang.org/issues/10724

**Comment by deivid-rodriguez** [comment]
Hi again @timscott, I've found a minimal program to reproduce your issue.

    class A
      def initialize
        loop { return }
      end
    end

    A.new
    puts "'next' should stop here, but it doesn't...."

Hopefully it'll be fixed soon! :)

**Comment by deivid-rodriguez** [comment]
Fixed in ruby-core! :smile:

**Comment by deivid-rodriguez** [comment]
This has been backported to Ruby 2.2 and Ruby 2.1, but won't be fixed for 2.0.

After 2.1.7 and 2.2.3 are released, I´ll start recommending those versions in the README and close this issue.

**Comment by deivid-rodriguez** [comment]
Here you go, closing!
